scarb/src/ops/package.rs

```diff
@@ -246,6 +262,42 @@ fn prepare_archive_recipe(pkg: &Package, opts: &PackageOpts) -> Result<ArchiveRe
         contents: ArchiveFileContents::OnDisk(pkg.manifest_path().to_owned()),
     });
 
+    if pkg.is_cairo_plugin() && !pkg.manifest.targets.iter().any(is_builtin) {
+        // Package crate with Cargo.
+        package_crate(pkg, opts, ws)?;
+
+        let crate_archive_basename = get_crate_archive_basename(pkg);
+        if crate_archive_basename != pkg.id.tarball_basename() {
+            ws.config().ui().warn(formatdoc!(
+                r#"
+                package name or version mismatch between cargo and scarb manifest
+                cargo: `{cargo_basename}`, scarb: `{scarb_basename}`
```

Comment:
I suggest 
```suggestion
                package name or version differs between Cargo manifest and Scarb manifest
                Scarb manifest: `{scarb_basename}`, Cargo manifest: `{cargo_basename}`
                these might be required to match in future Scarb releases 
```

Reply:
```suggestion
                package name or version differs between Cargo manifest and Scarb manifest
                Scarb manifest: `{scarb_basename}`, Cargo manifest: `{cargo_basename}`
                this might become an error in future Scarb releases 
```

WDYT?

Reply:
👌 great